types/lsp.ts

```diff
@@ -0,0 +1,8 @@
+export { TextDocument, Position } from 'vscode-languageserver-textdocument'
```

Comment:
If we aim to have pure `types` package, we shouldn't re-export anything from vscode packages other than `vscode-languageserver-types`. 
'vscode-languageserver-textdocument' and 'vscode-languageserver-protocol' contain implementation of whole LSP protocol, which is re-exported here, but that defies the purpose of `types` package.

Reply:
Makes sense, we were importing many types directly from `vscode-languageserver-protocol`. Though I'm not sure about `vscode-languageserver-textdocument` as an implementation of LSP protocol, it seems to only define text document specific types which would fall under a "pure types" package criteria

Reply:
vscode themselves seem to use `textdocument` to complement their `types` package https://github.com/microsoft/vscode-languageserver-node/blob/main/types/src/main.ts#L4517C1-L4518C1